src/main/java/com/shapesecurity/shift/es2016/semantics/Explicator.java

```diff
@@ -119,77 +120,56 @@
 	final MultiHashTable<FunctionBody, WithStatement> withStatementsInFunctions;
 	@Nonnull
 	final F<ImmutableList<Directive>, Boolean> isCandidateForInlining;
-
-	Explicator(@Nonnull Script script) {
-		this.program = Either.left(script);
-		this.scope = ScopeAnalyzer.analyze(script);
-		this.jumpMap = FinallyJumpReducer.analyze(script);
-		this.scopeLookup = new ScopeLookup(this.scope);
-		this.functionScopes = new IdentityHashMap<>();
-		this.withReferences = findWithReferences(this.scope);
-		this.withObjects = new IdentityHashMap<>();
-		this.withStatementsInFunctions = FindWithsReducer.reduce(Either.extract(program)).right;
-		this.isCandidateForInlining = list -> false;
-	}
-
-	Explicator(@Nonnull Module module) {
-		this.program = Either.right(module);
-		this.scope = ScopeAnalyzer.analyze(module);
-		this.jumpMap = FinallyJumpReducer.analyze(module);
-		this.scopeLookup = new ScopeLookup(this.scope);
-		this.functionScopes = new IdentityHashMap<>();
-		this.withReferences = findWithReferences(this.scope);
-		this.withObjects = new IdentityHashMap<>();
-		this.withStatementsInFunctions = FindWithsReducer.reduce(Either.extract(program)).right;
-		this.isCandidateForInlining = list -> false;
-	}
-
-	Explicator(@Nonnull Script script, @Nonnull F<ImmutableList<Directive>, Boolean> isCandidateForInlining) {
-		this.program = Either.left(script);
-		this.scope = ScopeAnalyzer.analyze(script);
-		this.jumpMap = FinallyJumpReducer.analyze(script);
+	final boolean throwOnEval;
+
+	Explicator(@Nonnull Program program, @Nonnull F<ImmutableList<Directive>, Boolean> isCandidateForInlining, boolean throwOnEval) {
```

Comment:
Let's have additional constructors that give defaults for the recently-added parameters.

Reply:
The constructor is not public. There's overloads for the public static methods; that seems sufficient.